### PR TITLE
[backend] Switching get exercises to raw native query

### PR DESCRIPTION
**Pull request by Dimfacion**
### Proposed changes

* Switching get exercises to raw native query
*

### Related issues

*
*

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: openbas-model/src/main/java/io/openbas/database/repository/InjectRepository.java [hunk 1]
```diff
@@ -102,4 +103,20 @@ void importSaveForScenario(@Param("id") String id,
...
+          "coalesce(array_agg(assets.asset_id) FILTER ( WHERE assets.asset_id IS NOT NULL ), '{}') as inject_assets, " +
+          "coalesce(array_agg(iag.asset_group_id) FILTER ( WHERE iag.asset_group_id IS NOT NULL ), '{}') as inject_asset_groups, " +
+          "coalesce(array_agg(ie.inject_expectation_id) FILTER ( WHERE ie.inject_expectation_id IS NOT NULL ), '{}') as inject_expectations " +
+          "FROM injects " +
+          "LEFT JOIN injects_teams it ON injects.inject_id = it.inject_id " +
+          "LEFT JOIN injects_assets ia ON injects.inject_id = ia.inject_id " +
+          "LEFT JOIN injects_asset_groups iag ON injects.inject_id = iag.inject_id " +
+          "LEFT JOIN asset_groups_assets aga ON aga.asset_group_id = iag.asset_group_id " +
+          "LEFT JOIN assets ON assets.asset_id = ia.asset_id OR aga.asset_id = assets.asset_id " +
+          "LEFT JOIN injects_expectations ie ON injects.inject_id = ie.inject_id " +
```
Comment (RomuDeuxfois) [inline openbas-model/src/main/java/io/openbas/database/repository/InjectRepository.java:118]: I am not sure that we need to retrieve asset from asset group

File: openbas-model/src/main/java/io/openbas/database/repository/ExerciseRepository.java [hunk 2]
```diff
@@ -88,4 +89,35 @@ public interface ExerciseRepository extends CrudRepository<Exercise, String>,
...
+            "WHERE users_groups.user_id = :userId " +
+            "GROUP BY ex.exercise_id ;", nativeQuery = true)
+    List<RawExercise> rawAllGranted(@Param("userId") String userId);
+
+    @Query(value = "SELECT ie.inject_expectation_type, ie.inject_expectation_score " +
+            "FROM injects_expectations ie " +
+            "INNER JOIN injects ON ie.inject_id = injects.inject_id " +
+            "INNER JOIN exercises ON injects.inject_exercise = exercises.exercise_id " +
+            "WHERE exercises.exercise_created_at < :from ;", nativeQuery = true)
+    List<RawInjectExpectation> rawInjectOfExerciceList(@Param("from") Instant from);
```
[the remark this answers is not in the document]
Comment (Dimfacion) [reply openbas-model/src/main/java/io/openbas/database/repository/ExerciseRepository.java:122]: Good point (y)
I'm removing it

File: openbas-model/src/main/java/io/openbas/database/repository/InjectRepository.java (hunk 1 above)
Comment (Dimfacion) [reply openbas-model/src/main/java/io/openbas/database/repository/InjectRepository.java:118]: Unfortunatelly, we do. It's used in AtomicTestingUtils.java line 103 (and more after this line)

File: openbas-model/src/main/java/io/openbas/database/raw/impl/SimpleRawInjectExpectation.java [hunk 3]
```diff
@@ -17,11 +21,47 @@ public Integer getInject_expectation_score() {
...
+
     public void setInject_expectation_type(String inject_expectation_type) {
         this.inject_expectation_type = inject_expectation_type;
     }
 
     public void setInject_expectation_score(Integer inject_expectation_score) {
         this.inject_expectation_score = inject_expectation_score;
     }
+
+    public void setTeam_id(String team_id) {
```
Comment (RomuDeuxfois) [inline openbas-model/src/main/java/io/openbas/database/raw/impl/SimpleRawInjectExpectation.java:52]: You can use lombok on it

File: openbas-model/src/main/java/io/openbas/database/raw/RawInjectExpectation.java [hunk 4]
```diff
@@ -6,4 +6,10 @@ public interface RawInjectExpectation {
 
     Integer getInject_expectation_score();
 
+    String getTeam_id();
```
Comment (RomuDeuxfois) [inline openbas-model/src/main/java/io/openbas/database/raw/RawInjectExpectation.java:9]: If I understand correclty, we have one interface for two different queries ?
So, all the properties will not be filled depending on wich query is used ?

File: openbas-model/src/main/java/io/openbas/database/repository/ExerciseRepository.java [hunk 5]
```diff
@@ -88,4 +111,37 @@ public interface ExerciseRepository extends CrudRepository<Exercise, String>,
...
+    /**
+     *  Get the raw version of the exercises
+     * @return the list of exercises
+     */
+    @Query(value = " SELECT ex.exercise_category, ex.exercise_id, ex.exercise_status, ex.exercise_start_date, ex.exercise_name, " +
+            " ex.exercise_subtitle, array_agg(et.tag_id) FILTER ( WHERE et.tag_id IS NOT NULL ) as exercise_tags, " +
+            " array_agg(injects.inject_id) FILTER ( WHERE injects.inject_id IS NOT NULL ) as inject_ids " +
+            "FROM exercises ex " +
+            "LEFT JOIN injects_expectations ie ON ex.exercise_id = ie.exercise_id " +
+            "LEFT JOIN injects ON ie.inject_id = injects.inject_id " +
```
Comment (RomuDeuxfois) [inline openbas-model/src/main/java/io/openbas/database/repository/ExerciseRepository.java:124]: I didn't know you could omit the variable name from the table
LEFT JOIN injects X ON ...

File: openbas-model/src/main/java/io/openbas/database/repository/InjectRepository.java [hunk 6]
```diff
@@ -102,4 +103,20 @@ void importSaveForScenario(@Param("id") String id,
...
+          "WHERE injects.inject_id IN :ids " +
+          "GROUP BY injects.inject_id;", nativeQuery = true)
+  List<RawInject> findRawByIds(@Param("ids")List<String> ids);
```
Comment (RomuDeuxfois) [inline openbas-model/src/main/java/io/openbas/database/repository/InjectRepository.java:121]: I don't see the variable for getAsset_group_id in RawInject interface

**Comment by RomuDeuxfois** [comment]
Tested 🆗

File: openbas-model/src/main/java/io/openbas/database/raw/impl/SimpleRawInjectExpectation.java (hunk 3 above)
Comment (Dimfacion) [reply openbas-model/src/main/java/io/openbas/database/raw/impl/SimpleRawInjectExpectation.java:52]: Good point ! Doing

File: openbas-model/src/main/java/io/openbas/database/raw/RawInjectExpectation.java (hunk 4 above)
Comment (Dimfacion) [reply openbas-model/src/main/java/io/openbas/database/raw/RawInjectExpectation.java:9]: Yes, it's like a classic instanciation of an object with unused properties being set to their default value.

File: openbas-model/src/main/java/io/openbas/database/repository/InjectRepository.java (hunk 6 above)
Comment (Dimfacion) [reply openbas-model/src/main/java/io/openbas/database/repository/InjectRepository.java:121]: It's on line 9 of RawInject.java

File: openbas-model/src/main/java/io/openbas/database/raw/impl/SimpleRawInjectExpectation.java (hunk 3 above)
Comment (Dimfacion) [reply openbas-model/src/main/java/io/openbas/database/raw/impl/SimpleRawInjectExpectation.java]: Done

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**